streaming/src/main/scala/org/apache/spark/streaming/ui/StreamingTab.scala

```diff
@@ -38,10 +38,14 @@ private[spark] class StreamingTab(ssc: StreamingContext)
   parent.attachTab(this)
 }
 
-private object StreamingTab {
+private[spark] object StreamingTab {
   def getSparkUI(ssc: StreamingContext): SparkUI = {
     ssc.sc.ui.getOrElse {
       throw new SparkException("Parent SparkUI to attach this tab to not found!")
     }
   }
+  
```

Comment:
Can you move this code into the StreamingTab class? The code is related to the tab instance, and so should be in the class. You could argue that `getSparkUI` should also be in the class, however it needs to be called from constructor (line 31) and hence cannot be in the class. But the `detach` function should be in the class.